src/core/app/authHooks.ts

```diff
@@ -1,58 +1,25 @@
 import { appConfig } from '~core/app_config';
-import { currentUserAtom, featureFlagsAtom } from '~core/shared_state';
+import { currentUserAtom } from '~core/shared_state';
 import { yandexMetrics } from '~core/metrics';
-import { loadFeatures, setFeatures } from './features';
+import { setFeatures } from './features';
 import type { AuthSuccessResponse } from '~core/auth/client/AuthClient';
-import type { CurrentUser } from './user';
-import type { ApiClient } from '~core/api_client';
 
-// TODO: rework unauthenticated flow and naming
-export async function onPublicLogin(apiClient: ApiClient) {
-  // load features for public user, without token
-  const featuresResponse = await loadFeatures(apiClient, false);
-  setFeatures(featuresResponse);
-}
-
-export async function onLogin(apiClient: ApiClient, response: AuthSuccessResponse) {
-  const jwtUserdata = {
-    username: response.jwtData.preferred_username,
-    token: response.token,
-    email: response.jwtData.email,
-    firstName: response.jwtData.given_name,
-    lastName: response.jwtData.family_name,
-    id: response.jwtData.sub,
-  };
-
-  externalLoginTasks(jwtUserdata);
-
-  // load profile
-  const profileResponse = apiClient.get<CurrentUser>(
-    '/users/current_user',
-    undefined,
-    true,
-  );
-
-  // load user features
-  const featuresResponse = loadFeatures(apiClient, true);
-
-  const [profileSettled, featuresSettled] = await Promise.allSettled([
-    profileResponse,
-    featuresResponse,
-  ]);
-  if (profileSettled.status === 'fulfilled' && profileSettled.value) {
-    const mergedUserdata = {
-      ...profileSettled.value,
-      id: jwtUserdata.id,
-      token: jwtUserdata.token,
-    };
-    currentUserAtom.setUser.dispatch(mergedUserdata);
-  }
-  if (featuresSettled.status === 'fulfilled' && featuresSettled.value) {
-    setFeatures(featuresSettled.value);
+export async function onLogin(response?: AuthSuccessResponse) {
+  if (response) {
+    if (appConfig.user) {
+      externalLoginTasks(appConfig.user);
+      const mergedUserdata = {
+        ...appConfig.user,
+        id: response.jwtData.sub,
+        token: response.token,
+      };
+      currentUserAtom.setUser.dispatch(mergedUserdata);
+    }
   }
+  setFeatures(appConfig.effectiveFeatures);
 }
```

Comment:
1. response is now optional, but there are no actions when it's not provided - weird
2. `setFeatures` called even when no response and config didn't change, is that expected?
3. if above thigs expected why not remove extra nest level
```suggestion
export async function onLogin(response?: AuthSuccessResponse) {
  if (response && appConfig.user) {
    externalLoginTasks(appConfig.user);
    const mergedUserdata = {
      ...appConfig.user,
      id: response.jwtData.sub,
      token: response.token,
    };
    currentUserAtom.setUser.dispatch(mergedUserdata);
  }
  setFeatures(appConfig.effectiveFeatures);
```

Reply:
1 - now it combines logic for auth and anon init
2 yes, to be refactored in other tasks
3 yes, thanks